Cake.Console/Internals/CakeContainer.cs

```diff
@@ -42,7 +43,12 @@ public CakeContainer(CakeConsoleArguments args)
             services.Add(Builder.Singleton<IHostBuilderBehaviour, TaskRegisteringBehaviour>());
 
             // the host
-            services.Add(Builder.Singleton<IScriptHost, CakeHost>());
+            var host = args.HasArgument("dryrun") ? typeof(DryRunScriptHost) :
+                args.HasArgument("description") ? typeof(DescriptionScriptHost) :
+                args.HasArgument("tree") ? typeof(TreeScriptHost) :
+                typeof(CakeHost);

```

Comment:
Maybe an switch expression is more readable here?
```suggestion
            var host =
                (
                    DryRunScriptHost: args.HasArgument("dryrun"),
                    DescriptionScriptHost: args.HasArgument("description"),
                    TreeScriptHost: args.HasArgument("tree")
                ) switch {
                    { DryRunScriptHost: true }      => typeof(DryRunScriptHost),
                    { DescriptionScriptHost: true } => typeof(DescriptionScriptHost),
                    { TreeScriptHost: true }        => typeof(TreeScriptHost),
                    _                               => typeof(CakeHost)
                };
```

Reply:
In all candor, I disagree

Reply:
Totally ok :)